src/contrib/persistence/Akka.Persistence.Sql.Common/Journal/QueryExecutor.cs

```diff
@@ -628,8 +628,8 @@ protected DbCommand GetCommand(DbConnection connection, string sql)
         protected virtual void WriteEvent(DbCommand command, IPersistentRepresentation e, IImmutableSet<string> tags)
         {
             var manifest = string.IsNullOrEmpty(e.Manifest) ? e.Payload.GetType().TypeQualifiedName() : e.Manifest;
-            var serializer = Serialization.FindSerializerFor(e.Payload);
-            var binary = serializer.ToBinary(e.Payload);
+            var serializer = Serialization.FindSerializerForType(typeof(IPersistentRepresentation));
```

Comment:
One problem a co-worker pointed out though, is this will break existing users of the sql plugin on an upgrade.  I still think the serialization changes are good to pick up and leverage moving forward, but I'll go ahead and rework the SQL changes specifically.  Moving forward it'd be worth considering using the new model and perhaps we can create a plugin that has breaking changes.